### ISSUE TITLE
Bug: StringLiterals gives false positive

### ISSUE THREAD

**Issue by perlun** [bbatsov/rubocop#5797]
This code will give a Rubocop warning (v0.55.0):

```ruby
openssl "genrsa -out ""#{path}"" 2048"
```

The warning is `C: Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols. (https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)`

It seems like Rubocop doesn't understand that "" continues the string, so it thinks that `genrsa out ` is the full string, where no interpolation is indeed needed.

**Comment by mikegee** [comment]
You're using 3 adjacent strings with implied concatenation. Rubocop wants you to use single quotes on the first and third.

`rubocop --auto-correct` changes it to:

```ruby
openssl 'genrsa -out '"#{path}"' 2048'
```

**Comment by perlun** [comment]
Thanks. You might be right about this; I must admit I haven't studied the details. Another way to phrase it:

`openssl "genrsa -out \"#{path}\" 2048"`

That way, a single string and Rubocop will likely accept it. Thanks for the feedback, this is as you suggest a non-bug in Rubocop. Closing.

**Comment by bbatsov** [comment]
Alternatively you could have used `format` here (that's what I would have done).

**Comment by perlun** [comment]
@bbatsov Good suggestion. For reference, I ended up with something like:

```ruby
        parameters_string = sprintf('genrsa -out "%<path>s" 2048', path: path)
        openssl parameters_string
```

(`sprintf == format`. More details [in the Style Guide](https://github.com/bbatsov/ruby-style-guide#sprintf).)

**Comment by mikegee** [comment]
> Another way to phrase it:
> `openssl "genrsa -out \"#{path}\" 2048"`

This gives different a different result. There are no double quotes in the string given to `openssl` in your original example.

**Comment by perlun** [comment]
@mikegee 

> This gives a different result. There are no double quotes in the string given to openssl in your original example.

Indeed. It was a silly mistake by me, converting some [TypeScript code](https://github.com/davewasmer/devcert/blob/master/src/certificates.ts#L44) to Ruby... It works without the double quotes unless your path contains embedded spaces. But you're definitely right, thanks for point that out.

@Drenmi 

> Hm. We don't have a cop that checks implicit string concatenation? 🤔

Would make sense. I _thought_ I was doing something else than I apparently was... 😆
